lib/src/navigation/routes/observers/route_observer.dart

```diff
@@ -49,124 +105,124 @@ class GetObserver extends NavigatorObserver {
   String current;
   String previous;
   dynamic args;
-
-  // String previousArgs;
   String removed;
 
-  String name(Route<dynamic> route) {
-    if (route?.settings?.name != null) {
-      return route.settings.name;
-    } else if (route is GetPageRoute) {
-      return route.routeName;
-    } else if (route is GetDialogRoute) {
-      return route.name;
-    } else if (route is GetModalBottomSheetRoute) {
-      return route.name;
-    } else {
-      return route?.settings?.name;
-    }
-  }
+  // FIXME: this should be public?
+  String name(Route<dynamic> route) => _extractRouteName(route);
 
   @override
-  void didPush(Route<dynamic> route, Route<dynamic> previousRoute) {
-    var isGetPageRoute = route is GetPageRoute;
-    var isSnackbar = route is SnackRoute;
-    var isDialog = route is GetDialogRoute;
-    var isBottomSheet = route is GetModalBottomSheetRoute;
-    var routeName = name(route);
-
-    if (isSnackbar) {
-      GetConfig.log("OPEN SNACKBAR $routeName");
-    } else if (isBottomSheet || isDialog) {
-      GetConfig.log("OPEN $routeName");
-    } else if (isGetPageRoute) {
-      GetConfig.log("GOING TO ROUTE $routeName");
+  void didPush(Route route, Route previousRoute) {
+    super.didPush(route, previousRoute);
+    final newRoute = _RouteMetaData.ofRoute(route);
+
+    if (newRoute.isSnackbar) {
+      GetConfig.log("OPEN SNACKBAR ${newRoute.name}");
+    } else if (newRoute.isBottomSheet || newRoute.isDialog) {
+      GetConfig.log("OPEN ${newRoute.name}");
+    } else if (newRoute.isGetPageRoute) {
+      GetConfig.log("GOING TO ROUTE ${newRoute.name}");
     }
-    GetConfig.currentRoute = routeName;
+    GetConfig.currentRoute = newRoute.name;
 
     _routeSend?.update((value) {
-      if (route is PageRoute) value.current = routeName;
+      // FIXME: why the 'is'? should't we always use 'newName'?
```

Comment:
Somente PageRoutes devem mudar o "currentRoute". O nome da Snackbar, dialogs, e bottomSheets não devem alterar essa propriedade. 
"value.current" é usado como "currentRoute". Acima você vai encontrar `GetConfig.currentRoute` que terá o nome alterado para "reference", e será removido dessa classe como um static. `GetConfig.currentRoute ` é usado pelo GetInstanceManager para saber à qual rota ou dependencia ele será linkado, para usar como "valvula de segurança", se o dispose de uma página falhar (muita gente usa setState dentro de um Builder com 2 widgets, ex: logged? Home():Login(), e isso não chama o dispose do widget quando o setState é chamado, então isso funciona como uma valvula de segurança para disposar controladores ainda se o dispose não for chamado). GetConfig.currentRoute armazena o hashcode de dialogs/bottomsheets, o nome da rota em caso de rota nomeada, e o nome do widget/page em caso de rotas não nomeadas, para remover eles, caso existem após o dispose da rota. 
Já value.current é usado pelo Get.currentRoute, que é usado publicamente para saber qual a rota atual (que deve ser uma PageRoute)

Reply:
Entendo o if enton, usar o newName não faria diferença, mas vou não vou mexer nisso então.
Vou deixar comentado no if que é explicito o is

Reply:
tem uns lugares que tão usando direto o `previousRoute?.settings?.name` em vez de usar o metodo de extrair o nome da rota, isso ta certo? 🤔

Reply:
agr fico em duvida se é correto usar `name` no `didRemove` remove por ex

Reply:
Pode usar o name, isso deveria ser padronizado, mas acabei fazendo modificações conforme a necessidade e virou essa bagunça auehauhea, obrigado por pegar isso também

Reply:
Oks, padronizei e use o (agora privado) `_extractRouteName` onde tinha uso de settings.name